core/trino-main/src/test/java/io/trino/sql/planner/TestRemoveRedundantJoinPlan.java

```diff
@@ -0,0 +1,68 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.sql.planner;
+
+import io.trino.sql.planner.assertions.BasePlanTest;
+import io.trino.sql.planner.plan.JoinNode;
+import io.trino.sql.tree.GenericLiteral;
+import org.testng.annotations.Test;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+import static io.trino.sql.planner.assertions.PlanMatchPattern.aggregation;
+import static io.trino.sql.planner.assertions.PlanMatchPattern.filter;
+import static io.trino.sql.planner.assertions.PlanMatchPattern.join;
+import static io.trino.sql.planner.assertions.PlanMatchPattern.output;
+import static io.trino.sql.planner.assertions.PlanMatchPattern.project;
+import static io.trino.sql.planner.assertions.PlanMatchPattern.values;
+import static io.trino.sql.planner.plan.JoinNode.DistributionType.REPLICATED;
+import static io.trino.sql.planner.plan.JoinNode.Type.INNER;
+
+public class TestRemoveRedundantJoinPlan
+        extends BasePlanTest
+{
+    @Test
+    public void testSemiJoinWithEmptyRight()
+    {
+        assertPlan("SELECT orderkey FROM orders WHERE orderkey IN (SELECT 1 WHERE false)",
+                output(List.of("orderkey"),
+                        values("orderkey")));
+    }
+
+    @Test
+    public void testSemiJoin()
```

Comment:
What this method is about?

Reply:
After add the RemoveRedundantJoin, I suppose to test the SemiJoin when the table  are not empty. it should be work as before.

Reply:
I don't think this test is needed, and also it would still require more work (eg i don't see https://github.com/trinodb/trino/pull/14335#discussion_r982044900 addressed). let's remove it